packages/core/src/components/table/table-input-wrapper/table-input-wrapper.tsx

```diff
@@ -0,0 +1,58 @@
+import { Component, Element, h, Host, State } from '@stencil/core';
+
+@Component({
+  tag: 'tds-table-input-wrapper',
+  styleUrl: 'table-input-wrapper.scss',
+  shadow: true,
+})
+export class TdsTableInputWrapper {
+  @Element() host: HTMLElement;
+
+  @State() renderSlot: boolean = true;
+
+  @State() inputFocused: boolean = false;
+
+  private handleSlotChange() {
+    this.validateSlot();
+  }
+
+  private validateSlot() {
+    const children = Array.from(this.host.children).filter(
+      (element) => element.tagName === 'INPUT',
+    ) as HTMLInputElement[];
+
+    if (children.length !== 1) {
+      console.warn('INPUT-WRAPPER: Wrapper only accepts input as children.');
+      this.renderSlot = false;
+    } else {
+      if (!this.renderSlot) this.renderSlot = true;
+
+      const input = children[0];
+      input.addEventListener('focus', () => {
+        this.inputFocused = true;
+      });
+      input.addEventListener('blur', () => {
+        this.inputFocused = false;
+      });
+    }
+  }
+
+  render() {
+    return (
+      <Host
+        class={{
+          'focused-input-wrapper': this.inputFocused,
+        }}
+      >
+        {this.renderSlot ? <slot onSlotchange={() => this.handleSlotChange()} /> : null}
+        <tds-icon
+          class="edit-icon"
+          slot="icon"
+          size="16px"
+          name="edit"
+          data-testid="tds-button-testid"
+        ></tds-icon>
```

Comment:
As the component will be used for column search too, maybe we should add the "magnifier" icon as a prefix too and then connect it to the some prop where user can define if it is search or edit. The default state would be undefined, so input without any icons. We can speak more about that tomorrow.

Reply:
As we discussed, we will create a separate component to be used for header cells and rename this one to be used for body cells.